gateway/utils/graphql/read.go

```diff
@@ -294,80 +359,102 @@ func (graph *Module) extractSelectionSet(field *ast.Field, dbAlias, col string,
 }
 
-func extractAggregate(ctx context.Context, v *ast.Field, store utils.M) (map[string][]string, error) {
+func extractAggregate(ctx context.Context, v *ast.Field, store utils.M, dbType, col string) (map[string][]string, error) {
 	functionMap := make(map[string][]string)
-	aggregateFound := false
+	aggregateFound := new(bool)
 	if v.SelectionSet == nil {
 		return nil, nil
 	}
 	for _, selection := range v.SelectionSet.Selections {
-		field := selection.(*ast.Field)
-
-		// Check if aggregate was found in the directive
-		if len(field.Directives) > 0 && field.Directives[0].Name.Value == "aggregate" {
-			// Get the required parameters
-			returnField := field.Name.Value
-			op, fieldName, err := getAggregateArguments(field.Directives[0], store)
-			if err != nil {
-				return nil, err
+		f, err := aggregateSingleField(ctx, selection.(*ast.Field), store, col, model.DBType(dbType), aggregateFound)
+		if err != nil {
+			return nil, err
+		}
+		for key, value := range f {
+			v, ok := functionMap[key]
+			if !ok {
+				functionMap[key] = value
+			} else {
+				functionMap[key] = append(v, value...)
 			}
+		}
+	}
+	return functionMap, nil
+}
 
+func aggregateSingleField(ctx context.Context, field *ast.Field, store utils.M, tableName string, dbType model.DBType, aggregateFound *bool) (map[string][]string, error) {
+	functionMap := make(map[string][]string)
+
+	// Check if aggregate was found in the directive
+	if len(field.Directives) > 0 && field.Directives[0].Name.Value == "aggregate" {
+		// Get the required parameters
+		returnField := field.Name.Value
+		op, fieldName, err := getAggregateArguments(field.Directives[0], store)
+		if err != nil {
+			return nil, err
+		}
+
+		if dbType != model.Mongo {
 			if fieldName == "" {
-				fieldName = returnField
+				fieldName = tableName + "." + returnField
 			}
-
-			colArray, ok := functionMap[op]
-			if !ok {
-				colArray = []string{}
+			if arr := strings.Split(fieldName, "."); len(arr) <= 1 {
+				fieldName = tableName + "." + fieldName
 			}
-
-			columnName := fmt.Sprintf("%s:%s:table", returnField, strings.Join(strings.Split(fieldName, "__"), "."))
-			colArray = append(colArray, columnName)
-			functionMap[op] = colArray
-			continue
+		} else if fieldName == "" {
+			fieldName = returnField
 		}
 
-		if field.Name.Value != utils.GraphQLAggregate || field.SelectionSet == nil {
-			continue
+		colArray, ok := functionMap[op]
+		if !ok {
+			colArray = []string{}
 		}
 
-		if aggregateFound {
-			return nil, helpers.Logger.LogError(helpers.GetRequestID(ctx), "GraphQL query cannot have multiple aggregate fields, specify all functions in single aggregate field", nil, nil)
+		columnName := fmt.Sprintf("%s:%s:table", returnField, strings.Join(strings.Split(fieldName, "__"), "."))
+		colArray = append(colArray, columnName)
+		functionMap[op] = colArray
+		return functionMap, nil
+	}
+
+	if field.Name.Value != utils.GraphQLAggregate || field.SelectionSet == nil {
```

Comment:
We should deprecate this aggregate syntax. Create an issue to remove this syntax